packages/nextjs/hooks/scaffold-eth/useScaffoldContractRead.ts

```diff
@@ -10,23 +11,30 @@ import scaffoldConfig from "~~/scaffold.config";
  * @param args - args to be passed to the function call
  * @param readConfig - extra wagmi configuration
  */
-export const useScaffoldContractRead = <TReturn = any>(
-  contractName: string,
-  functionName: string,
-  args?: any[],
+export const useScaffoldContractRead = <
+  TContractName extends ContractName,
+  TFunctionName extends ExtractAbiFunctionNames<ContractAbi<TContractName>, "pure" | "view">,
+>(
+  contractName: TContractName,
+  functionName: TFunctionName,
+  args?: AbiFunctionArguments<ContractAbi<TContractName>, TFunctionName>,
   readConfig?: Parameters<typeof useContractRead>[0],
 ) => {
-  const { data: deployedContractData } = useDeployedContractInfo(contractName);
+  const { data: deployedContract } = useDeployedContractInfo(contractName);
 
   return useContractRead({
     chainId: scaffoldConfig.targetNetwork.id,
     functionName,
-    address: deployedContractData?.address,
-    abi: deployedContractData?.abi as Abi,
+    address: deployedContract?.address,
+    abi: deployedContract?.abi,
     watch: true,
-    args,
+    args: args as unknown[],
     ...readConfig,
-  }) as Omit<ReturnType<typeof useContractRead>, "data"> & {
-    data: TReturn;
+  }) as Omit<ReturnType<typeof useContractRead>, "data" | "refetch"> & {
+    data: AbiFunctionReturnType<ContractAbi, TFunctionName>;
```

Comment:
I think we should also union  it with undefined : 
```ts
data: AbiFunctionReturnType<ContractAbi, TFunctionName> | undefined;
```

Reason : 
It will be `undefined` initially and may give false information to the developer in compile time and he will get a runtime error if he won't add check for `undefined`. 

Even wagmi has it : 
![Screenshot 2023-03-20 at 8 14 19 PM](https://user-images.githubusercontent.com/80153681/226375338-76416257-0b2c-4669-b8f0-eec130405e4a.jpg)


Sorry, I know its not right time to review it 😅 since it's still in the `draft` stage. But just noticed while playing with hooks and wanted to mention it so that I don't forget it 🙌